create_data_token() fail when using default CONFIG_FILE value, looks like artifacts issue

**Describe the bug**

The following configuration fails:
- using barge
- import ocean.py (vs git clone)
- a config.ini file that includes location of  addresses artifacts is _not_ specified
- exported CONFIG_FILE ennvar as config.ini
- in py code, do _not_ specify the location of config file (it's supposed to use CONFIG_FILE envvar)
- trying to do simple quickstart functionality 

Error: 
```console
  File "/home/trentmc/code/ocean.py/ocean_lib/ocean/ocean.py", line 157, in create_data_token
    assert address, "new datatoken has no address"
```

**To Reproduce**

1. Do initial steps in developers.md to install dependencies, run the services, set up contracts

2. In console:
```console
export CONFIG_FILE=config.ini
```

(This is actually done as part of "set up contracts", but putting here just to make it obvious that it was set.)

3. In python console:

```python
import os
from ocean_lib.config import Config
from ocean_lib.ocean.ocean import Ocean
from ocean_lib.web3_internal.wallet import Wallet

private_key = os.getenv('TEST_PRIVATE_KEY1')
config = Config() # note how no config file is specified. In this case, it uses CONFIG_FILE envvar
ocean = Ocean(config)

print("create wallet: begin")
wallet = Wallet(ocean.web3, private_key=private_key)
print(f"create wallet: done. Its address is {wallet.address}")

print("create datatoken: begin.")
datatoken = ocean.create_data_token("Dataset name", "dtsymbol", from_wallet=wallet) 
```

The last line will fail. Traceback is below.

Note: it _does_ pass if the filename is explicitly stated. 

**Expected behavior**
The last line not to fail. While all pytest tests still passing.

Related: 
- #268 "Aquarius + tox crashes, looks like ocean.py artifacts issue"
- #185 "create_data_token() fail when addresses not specified, looks like artifacts issue" -- failed with the same error, with a different envvar setup (it didn't use config.ini, _this_ issue does)
- [multi-repo#83](https://github.com/oceanprotocol/multi-repo-issue/issues/83) "ocean-contracts is hard to use in python, including in ocean.py"



**Config.ini contents**
For reference, here is part of the config.ini file. 
```text
[eth-network]
network = ganache
artifacts.path = ~/.ocean/ocean-contracts/artifacts
address.file = ~/.ocean/ocean-contracts/artifacts/address.json
```

`artifacts.path` and `address.file` are the defaults for barge.

**Traceback**

```console
>>> datatoken = ocean.create_data_token("Dataset name", "dtsymbol", from_wallet=wallet) 
WARNING:root:No logs were found for tx 0x47c1f5e6096eb2d9a5cce8f7cd086b736b570ff5a3c576c4055579ec90df693d.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/trentmc/code/ocean.py/ocean_lib/ocean/ocean.py", line 157, in create_data_token
    assert address, "new datatoken has no address"
AssertionError: new datatoken has no address
```

Update: I just tried to reproduce it again. It now fails earlier, before even getting to the error reported in the description.

In step 3, in Python console, I make it to the following line.
```python
import os
from ocean_lib.config import Config
from ocean_lib.ocean.ocean import Ocean
```
>>> from ocean_lib.ocean.ocean import Ocean
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/trentmc/code/ocean.py/ocean_lib/ocean/ocean.py", line 13, in <module>
    from ocean_lib.data_provider.data_service_provider import DataServiceProvider
  File "/home/trentmc/code/ocean.py/ocean_lib/data_provider/data_service_provider.py", line 19, in <module>
    from ocean_lib.web3_internal.utils import add_ethereum_prefix_and_hash_msg
  File "/home/trentmc/code/ocean.py/ocean_lib/web3_internal/utils.py", line 85, in <module>
    def privateKeyToAddress(private_key: str) -> str:
  File "/home/trentmc/code/ocean.py/ocean_lib/enforce_typing_shim.py", line 18, in enforce_types_shim
    if not ConfigProvider.get_config()["util"].getboolean("typecheck"):
  File "/home/trentmc/code/ocean.py/ocean_lib/config_provider.py", line 17, in get_config
    raise AssertionError("set_config first.")
AssertionError: set_config first.
```

I will report this as a new issue.

--- a/ocean_lib/config.py
+++ b/ocean_lib/config.py
@@ -9,6 +9,7 @@ import os
 import site
 from pathlib import Path
 
+from ocean_lib.ocean.env_constants import ENV_CONFIG_FILE
 from ocean_lib.web3_internal.constants import GAS_LIMIT_DEFAULT
 
 DEFAULT_NETWORK_HOST = "localhost"
@@ -124,7 +125,7 @@ class Config(configparser.ConfigParser):
         self._web3_provider = None
         self._section_name = "eth-network"
         self._logger = logging.getLogger("config")
-
+        
         if filename:
             self._logger.debug(f"Config: loading config file {filename}")
             with open(filename) as fp:
@@ -136,6 +137,12 @@ class Config(configparser.ConfigParser):
         elif options_dict:
             self._logger.debug(f"Config: loading from dict {options_dict}")
             self.read_dict(options_dict)
+        else:
+            filename = os.getenv(ENV_CONFIG_FILE)
+            self._logger.debug(f"Config: loading config file {filename}")
+            with open(filename) as fp:
+                text = fp.read()
+                self.read_string(text)
 
         self._load_environ()
 

--- a/ocean_lib/config.py
+++ b/ocean_lib/config.py
@@ -139,6 +139,8 @@ class Config(configparser.ConfigParser):
             self.read_dict(options_dict)
         else:
             filename = os.getenv(ENV_CONFIG_FILE)
+            if filename is None:
+                raise ValueError(f'Config file envvar "{ENV_CONFIG_FILE}" is empty')
             self._logger.debug(f"Config: loading config file {filename}")
             with open(filename) as fp:
                 text = fp.read()

--- a/ocean_lib/config.py
+++ b/ocean_lib/config.py
@@ -125,7 +125,7 @@ class Config(configparser.ConfigParser):
         self._web3_provider = None
         self._section_name = "eth-network"
         self._logger = logging.getLogger("config")
-        
+
         if filename:
             self._logger.debug(f"Config: loading config file {filename}")
             with open(filename) as fp:

--- a/ocean_lib/test/test_config_provider.py
+++ b/ocean_lib/test/test_config_provider.py
@@ -36,7 +36,7 @@ def test_metadataStoreUri_version():
     assert "http://ItWorked.com" == ocean_instance.config.aquarius_url
 
 
-def test_config__filename_given__file_doesnt_exist():
+def test_config_filename_given_file_doesnt_exist():
     """Test creating a Config object.
     Setup: filename given, file doesn't exist
     Expect: complain
@@ -48,7 +48,7 @@ def test_config__filename_given__file_doesnt_exist():
         config = Config(filename=config_file_name)
 
 
-def test_config__filename_given__file_exists__malformed_content(monkeypatch, tmp_path):
+def test_config_filename_given_file_exists_malformed_content(monkeypatch, tmp_path):
     """Test creating a Config object.
     Setup: filename given, file exists, malformed content
     Expect: complain
@@ -60,7 +60,7 @@ def test_config__filename_given__file_exists__malformed_content(monkeypatch, tmp
         config = Config()
 
 
-def test_config__filename_given__file_exists__wellformed_content():
+def test_config_filename_given_file_exists_wellformed_content():
     """Test creating a Config object.
     Setup: filename given, file exists, content is well-formed
     Expect: success
@@ -75,7 +75,7 @@ def test_config__filename_given__file_exists__wellformed_content():
     assert isinstance(config.gas_limit, int)
 
 
-def test_config__filename_not_given__envvar_is_empty(monkeypatch):
+def test_config_filename_not_given_envvar_is_empty(monkeypatch):
     """Test creating a Config object.
     Setup: filename not given, envvar is empty
     Expect: complain
@@ -85,7 +85,7 @@ def test_config__filename_not_given__envvar_is_empty(monkeypatch):
         config = Config()
 
 
-def test_config__filename_not_given__file_doesnt_exist(monkeypatch):
+def test_config_filename_not_given_file_doesnt_exist(monkeypatch):
     """Test creating a Config object.
     Setup: filename not given, default file doesn't exist
     Expect: complain
@@ -98,7 +98,7 @@ def test_config__filename_not_given__file_doesnt_exist(monkeypatch):
         config = Config()
 
 
-def test_config__filename_not_given__file_exists__malformed_content(
+def test_config_filename_not_given_file_exists_malformed_content(
     monkeypatch, tmp_path
 ):
     """Test creating a Config object.
@@ -112,7 +112,7 @@ def test_config__filename_not_given__file_exists__malformed_content(
         config = Config()
 
 
-def test_config__filename_not_given__file_exists__wellformed_content(monkeypatch):
+def test_config_filename_not_given_file_exists_wellformed_content(monkeypatch):
     """Test creating a Config object.
     Setup: filename not given, default file exists, content is well-formed
     Expect: success. Uses config file at ENV_CONFIG_FILE.
@@ -135,7 +135,7 @@ def _create_malformed_conffile(tmp_path):
     return config_file_name
 
 
-def test_config__from_text__wellformed_content():
+def test_config_from_text_wellformed_content():
     """Tests creating Config object.
     Setup: from raw text, content is well-formed
     Expect: success
@@ -148,7 +148,7 @@ def test_config__from_text__wellformed_content():
     assert config.aquarius_url == "https://another-aqua.url"
 
 
-def test_config__from_text__malformed_content():
+def test_config_from_text_malformed_content():
     """Tests creating Config object.
     Setup: from raw text, content is malformed
     Expect: complain

--- a/ocean_lib/test/test_config_provider.py
+++ b/ocean_lib/test/test_config_provider.py
@@ -3,8 +3,12 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
+import os.path
+import pytest
+
 from ocean_lib.config import Config
 from ocean_lib.config_provider import ConfigProvider
+from ocean_lib.ocean.env_constants import ENV_CONFIG_FILE
 from ocean_lib.example_config import ExampleConfig
 from ocean_lib.ocean.ocean import Ocean
 from tests.resources.ddo_helpers import get_resource_path
@@ -32,8 +36,35 @@ def test_metadataStoreUri_version():
     assert "http://ItWorked.com" == ocean_instance.config.aquarius_url
 
 
-def test_config_from_filename():
-    """Tests that a Config object can be set from a filename."""
+def test_config__filename_given__file_doesnt_exist():
+    """Test creating a Config object.
+    Setup: filename given, file doesn't exist
+    Expect: complain
+    """
+    config_file_name = "i_dont_exist.ini"
+    assert not os.path.exists(config_file_name)
+
+    with pytest.raises(Exception):
+        config = Config(filename=config_file_name)
+
+
+def test_config__filename_given__file_exists__malformed_content(monkeypatch, tmp_path):
+    """Test creating a Config object.
+    Setup: filename given, file exists, malformed content
+    Expect: complain
+    """
+    config_file_name = _create_malformed_conffile(tmp_path)
+
+    monkeypatch.setenv(ENV_CONFIG_FILE, config_file_name)
+    with pytest.raises(Exception):
+        config = Config()
+
+
+def test_config__filename_given__file_exists__wellformed_content():
+    """Test creating a Config object.
+    Setup: filename given, file exists, content is well-formed
+    Expect: success
+    """
     config_file_name = get_resource_path("config", "test_config.ini")
     config = Config(filename=config_file_name)
 
@@ -44,14 +75,87 @@ def test_config_from_filename():
     assert isinstance(config.gas_limit, int)
 
 
-def test_config_from_text():
-    """Tests that a Config object can be set from raw text."""
+def test_config__filename_not_given__envvar_is_empty(monkeypatch):
+    """Test creating a Config object.
+    Setup: filename not given, envvar is empty
+    Expect: complain
+    """
+    monkeypatch.delenv(ENV_CONFIG_FILE)
+    with pytest.raises(ValueError):
+        config = Config()
+
+
+def test_config__filename_not_given__file_doesnt_exist(monkeypatch):
+    """Test creating a Config object.
+    Setup: filename not given, default file doesn't exist
+    Expect: complain
+    """
+    config_file_name = "i_dont_exist.ini"
+    assert not os.path.exists(config_file_name)
+
+    monkeypatch.setenv(ENV_CONFIG_FILE, config_file_name)
+    with pytest.raises(Exception):
+        config = Config()
+
+
+def test_config__filename_not_given__file_exists__malformed_content(
+    monkeypatch, tmp_path
+):
+    """Test creating a Config object.
+    Setup: no filename given, default file exists, content is malformed
+    Expect: complain
+    """
+    config_file_name = _create_malformed_conffile(tmp_path)
+
+    monkeypatch.setenv(ENV_CONFIG_FILE, config_file_name)
+    with pytest.raises(Exception):
+        config = Config()
+
+
+def test_config__filename_not_given__file_exists__wellformed_content(monkeypatch):
+    """Test creating a Config object.
+    Setup: filename not given, default file exists, content is well-formed
+    Expect: success. Uses config file at ENV_CONFIG_FILE.
+    """
+    config_file_name = get_resource_path("config", "test_config.ini")
+    monkeypatch.setenv(ENV_CONFIG_FILE, str(config_file_name))
+
+    config = Config()
+
+    assert config.provider_address == "0x00bd138abd70e2f00903268f3db08f2d25677c9e"
+
+
+def _create_malformed_conffile(tmp_path):
+    """Helper function to some tests above. In: pytest tmp_pth. Out: str"""
+    d = tmp_path / "subdir"
+    d.mkdir()
+    config_file = d / "test_config_bad.ini"
+    config_file.write_text("Malformed content inside config file")
+    config_file_name = str(config_file)
+    return config_file_name
+
+
+def test_config__from_text__wellformed_content():
+    """Tests creating Config object.
+    Setup: from raw text, content is well-formed
+    Expect: success
+    """
     config_text = """
         [resources]
         aquarius.url = https://another-aqua.url
     """
     config = Config(text=config_text)
     assert config.aquarius_url == "https://another-aqua.url"
+
+
+def test_config__from_text__malfformed_content():
+    """Tests creating Config object.
+    Setup: from raw text, content is malformed
+    Expect: complain
+    """
+    config_text = "Malformed content inside config text"
+    with pytest.raises(Exception):
+        config = Config(text=config_text)
 
 
 def test_network_config():

--- a/ocean_lib/test/test_config_provider.py
+++ b/ocean_lib/test/test_config_provider.py
@@ -148,7 +148,7 @@ def test_config__from_text__wellformed_content():
     assert config.aquarius_url == "https://another-aqua.url"
 
 
-def test_config__from_text__malfformed_content():
+def test_config__from_text__malformed_content():
     """Tests creating Config object.
     Setup: from raw text, content is malformed
     Expect: complain

--- a/ocean_lib/test/test_config_provider.py
+++ b/ocean_lib/test/test_config_provider.py
@@ -98,9 +98,7 @@ def test_config_filename_not_given_file_doesnt_exist(monkeypatch):
         config = Config()
 
 
-def test_config_filename_not_given_file_exists_malformed_content(
-    monkeypatch, tmp_path
-):
+def test_config_filename_not_given_file_exists_malformed_content(monkeypatch, tmp_path):
     """Test creating a Config object.
     Setup: no filename given, default file exists, content is malformed
     Expect: complain

